File: j-LIB-core/src/main/java/io/github/dot166/jlib/registry/RegistryHelper.java
package io.github.dot166.jlib.registry;

import static io.github.dot166.jlib.registry.XmlHelper.readXmlFromFile;

import android.content.Context;

import java.util.List;
import java.util.Map;

public class RegistryHelper {

    public static List<Object> getFromRegistry(Context context) {
        return readXmlFromFile(context, "Registry.xml");
    }

    public static class Object {
        private String stationName;
        private String stationUrl;
        private String stationLogoUrl;

        public Object(Map<String, String> attributes) {
            stationName = attributes.get("objectName");
            stationUrl = attributes.get("objectUrl");
            stationLogoUrl = attributes.get("objectLogoUrl");
        }

        public String getLogoUrl() {
            return stationLogoUrl;
        }

        public String getName() {
            return stationName;
        }

        public String getUrl() {
            return stationUrl;
        }

        public void updateAttributes(Map<String, String> attributes) {
            stationName = attributes.get("objectName");
            stationUrl = attributes.get("objectUrl");
            stationLogoUrl = attributes.get("objectLogoUrl");
        }
    }
}

File: j-LIB-core/src/main/java/io/github/dot166/jlib/registry/RegistryHelper.kt
package io.github.dot166.jlib.registry

import android.content.Context
import io.github.dot166.jlib.registry.XmlHelper.readXmlFromFile

object RegistryHelper {
    fun getFromRegistry(context: Context): MutableList<Object> {
        return readXmlFromFile(context, "Registry.xml")
    }

    class Object(var name: String, var url: String, var logoUrl: String) {
        fun updateAttributes(name: String, url: String, logoUrl: String) {
            this.name = name
            this.url = url
            this.logoUrl = logoUrl
        }
    }
}
